### ISSUE TITLE
adding ability to change reported root dir file path

### ISSUE THREAD

**Issue by AdaskoTheBeAsT** [AriPerkkio/vitest-sonar-reporter#153]
Could you please add ability to define root dir file path similar to
https://github.com/sh33dafi/jest-sonar#readme

it is important in monorepos where frontend is not in main folder but in subfolder and sonar is not able to process result file in such case due to invalid file path

**Comment by AriPerkkio** [comment]
Could you provide some more information for this. Maybe an example of file structure, `vitest.config.ts` and what is expected to be on the report.

**Comment by AdaskoTheBeAsT** [comment]
ok imagine that solution folder contains subfolder frontend in which ui project is located
![image](https://github.com/AriPerkkio/vitest-sonar-reporter/assets/1914201/4c5e1081-33e6-48ac-a9ac-47516ea1cf9d)

due to fact that azure devops uses sonarqube/sonarcloud step and those steps works on root folder - reports are not properly consumed when given source files are not accessible by reported path
in that case report looks like this

![image](https://github.com/AriPerkkio/vitest-sonar-reporter/assets/1914201/e8cea924-7a35-4dfc-8977-271ace9cbb88)

it would be really nice if by using some additional parameter I can define root folder with value "..\" which in report it will be changed to 

![image](https://github.com/AriPerkkio/vitest-sonar-reporter/assets/1914201/2bc0275c-1a8b-4622-9464-140657b3ceae)

and then sonarqube/sonarcloud using that path will properly find source code and consume that test report

**Comment by AriPerkkio** [comment]
Thanks for the additional information. This feature request makes sense. I think I've run into that same exact problem myself couple of years ago with different test runner and reporter integration. Sonarqube can be tricky with relative paths.

To solve this I'm thinking about adding a new configuration option that users could use to modify the `path` properties. Some users could use this to append additional directories there (like in this one), and some could even use it to remove directories from the path.

In your case the configuration would look something like this:

```ts
test: {
    reporters: 'vitest-sonar-reporter',
    sonarReporterOptions: {
        onWriteFilePath(path: string) {
            // path is "apps/ui/src/app/app.spec.tsx"
            return `frontend/${path}`;
        }
    }
}
```

**Comment by AdaskoTheBeAsT** [comment]
hi @AriPerkkio 
thanks a lot taking a look into this - this will solve the problem with monorepo and sonarqube

in this case I could use 

```js
import * as path from 'path';
...

test: {
    reporters: 'vitest-sonar-reporter',
    sonarReporterOptions: {
        onWriteFilePath(reportPath: string) {
            // path is "apps/ui/src/app/app.spec.tsx"
            return path.relative("..", reportPath);
        }
    }
}
```

**Comment by AriPerkkio** [comment]
Yes, exactly like that.

I'll try to implement this in a week or so.

**Comment by AriPerkkio** [comment]
Looks like support from Vitest's side is needed first: https://github.com/vitest-dev/vitest/issues/5042

**Comment by AriPerkkio** [comment]
This is now included in `1.1.0` release.

**Comment by AdaskoTheBeAsT** [comment]
Thanks a lot!
